deploy/charts/operator/templates/registry-api-clusterrole.yaml

```diff
@@ -0,0 +1,27 @@
+{{- if .Values.operator.features.experimental }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
```

Comment:
`ClusterRole` or just `Role`?
IIRC, the `ListDeployedServers` method looks in the same namespace as the Registry

Reply:
can we know which namespace the Registry will be created in before deployment though? 

But I did realize we don't need the ConfigMap permissions since we are mounting the CM now, reducing the ClusterRole to only MCPRegistry.

Reply:
the current namespace should be either injected as an env-var/CLI flag or read from `/var/run/secrets/kubernetes.io/serviceaccount/namespace`

Reply:
yes, that's what we do (the latter, read from SA), but on the RBAC level we'd need to either programatically set the RBAC rules from the controller or know them beforehand.